sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/KeyEncryptionKeyClientBuilder.java

```diff
@@ -205,40 +243,50 @@ public KeyEncryptionKeyClientBuilder configuration(Configuration configuration)
      * newer version the client library will have the result of potentially moving to a newer service version.
      *
      * @param version {@link CryptographyServiceVersion} of the service to be used when making requests.
-     * @return The updated builder object.
+     *
+     * @return The updated {@link KeyEncryptionKeyClientBuilder} object.
      */
     public KeyEncryptionKeyClientBuilder serviceVersion(CryptographyServiceVersion version) {
         builder.serviceVersion(version);
+
         return this;
     }
 
     /**
-     * Sets the {@link RetryPolicy} that is used when each request is sent.
+     * Sets the {@link RetryPolicy} that is used when each request is sent. The default retry policy will be used in
+     * the pipeline, if not provided.
      *
-     * The default retry policy will be used in the pipeline, if not provided.
+     * @param retryPolicy User's retry policy applied to each request.
      *
-     * @param retryPolicy user's retry policy applied to each request.
-     * @return The updated KeyEncryptionKeyClientBuilder object.
-     * @throws NullPointerException if the specified {@code retryPolicy} is null.
+     * @return The updated {@link KeyEncryptionKeyClientBuilder} object.
+     *
+     * @throws NullPointerException If the specified {@code retryPolicy} is null.
      */
     public KeyEncryptionKeyClientBuilder retryPolicy(RetryPolicy retryPolicy) {
-        Objects.requireNonNull(retryPolicy, "The retry policy cannot be bull");
+        if (retryPolicy == null) {
+            throw logger.logExceptionAsError(new NullPointerException("'retryPolicy' cannot be null."));
+        }
```

Comment:
We generally don't log NPE mainly because we use `Objects.requireNonNull()`. It may not be a bad thing to do, just letting you know that it's not required.